src/common/string_util.cpp

```diff
@@ -266,6 +267,80 @@ string StringUtil::BytesToHumanReadableString(idx_t bytes, idx_t multiplier) {
 	return to_string(array[0]) + (bytes == 1 ? " byte" : " bytes");
 }
 
+idx_t StringUtil::ParseFormattedBytes(const string &arg) {
```

Comment:
Instead of throwing an exception in the function, could you implement this as:
```c++
bool StringUtil::TryParseFormattedBytes(const string &arg, idx_t result) {
    // ....
}

idx_t StringUtil::ParseFormattedBytes(const string &arg) {
    idx_t result;
    if (!TryParseFormattedBytes(arg, result)) {
        // throw
    }
}
```
This avoids having to do `try`/`catch` in other places, and the need to add the new exception class `NonNumericMemoryException`.

Reply:
Could return a `string` with the error message instead? If this string is `empty()`, parsing has succeded without errors. If a non-empty string is returned, we can throw a `ParserException` with the string

Reply:
Yes, we prefer to return a bool/enum/string that tells us whether the function was successful, rather than try/catch, especially for "shallow" function calls like this.

We do have some try/catch, but this is when the error you need to catch is deeply nested in a function call chain.